space-plugins/nuxt-base/composables/useAppBridge.ts

```diff
@@ -37,37 +19,108 @@ const getPostMessageAction = (type: PluginType): PostMessageAction => {
 	}
 };
 
-const useAppBridgeMessages = () => {
+const getParentHost = () => {
+	const storedHost = sessionStorage.getItem(KEY_PARENT_HOST);
+	if (storedHost) {
+		return storedHost;
+	}
+	const params = new URLSearchParams(location.search);
+	const protocol = params.get('protocol');
+	const host = params.get('host');
+	if (!protocol || !host) {
+		throw new Error('Missing `protocol` or `host` in query params');
+	}
+	return `${protocol}//${host}`;
+};
+
+const getSlug = () => {
+	const storedSlug = sessionStorage.getItem(KEY_SLUG);
+	if (storedSlug) {
+		return storedSlug;
+	}
+	const params = new URLSearchParams(location.search);
+	return params.get('slug');
+};
+
+const postMessageToParent = (payload: unknown) => {
+	window.parent.postMessage(payload, getParentHost());
+};
+
+const useAppBridgeAuth = ({
```

Comment:
could be also extracted as type `type UseAppBridge = (params: {authenticated: ()=> Promise<void>}) => void`

Reply:
but this is a nit pick so no need to implement